fe1-web/src/features/lao/hooks/LaoHooks.ts

```diff
@@ -0,0 +1,78 @@
+import FeatureContext from 'core/contexts/FeatureContext';
+import { useContext } from 'react';
+import { useSelector } from 'react-redux';
+import { LaoReactContext, LAO_FEATURE_IDENTIFIER } from '../interface';
+import { Lao } from '../objects';
+import {
+  selectCurrentLao,
+  selectCurrentLaoId,
+  selectIsLaoOrganizer,
+  selectLaosList,
+  selectLaosMap,
+} from '../reducer';
+
+export namespace LaoHooks {
+  /* Hooks passed by dependencies */
+
+  export const useLaoContext = (): LaoReactContext => {
+    const featureContext = useContext(FeatureContext);
+    // assert that the evoting context exists
+    if (!(LAO_FEATURE_IDENTIFIER in featureContext)) {
+      throw new Error('Lao context could not be found!');
+    }
+    return featureContext[LAO_FEATURE_IDENTIFIER] as LaoReactContext;
+  };
+
+  /**
+   * Gets the lao connection encoded in a way that can be shown in a QR code
+   * @returns The encoded lao connection ready to be rendered as a QR code
+   */
+  export const useEncodedLaoConnectionForQRCode = (
+    ...args: Parameters<LaoReactContext['encodeLaoConnectionForQRCode']>
```

Comment:
https://xkcd.com/297/ , but with TypeScript types instead of parentheses, and not quite as elegant 🤣

Reply:
😂😂

Fortunately I already changed this in another commit to
`export const useEncodeLaoConnectionForQRCode = () => useLaoContext().encodeLaoConnectionForQRCode;` :)